lms/views/lti/deep_linking.py

```diff
@@ -1,43 +1,56 @@
 """
-Views for handling deep linking launches.
+Deeplinking related views.
 
-A deep linking request is an LTI launch request (so it's a form
-submission containing all the usual launch params, including the OAuth 1
-signature) that's used by LMS's during the assignment creation process.
+If an LMS (like Canvas) is configured for "deep linking" then we need to send
+the results of the file picker to the LMS instead of storing it ourselves. This
+is done by the front-end.
 
-When an instructor creates a Hypothesis assignment in an LMS that supports
-deep linking the LMS launches our app in order for us to present an
-interface for the user to select a "content item" (in our case: a document to
-be annotated) for use with the assignment.
+When the LMS launches us with a deep linked assignment, we will get the
+document url as part of the launch params, instead of reading it from the DB in
+`module_item_configurations`.
 
-The spec requires that deep linking requests have an ``lti_message_type``
-parameter with the value ``ContentItemSelectionRequest``,
-but we don't actually require the requests to have this parameter: instead we
-use a separate URL to distinguish deep linking launches.
+The flow is:
 
-When the user selects a document we get the browser to POST the selection back
-to the LMS in a form submission with the ``lti_message_type`` parameter set to
-``ContentItemSelection``. The original ``ContentItemSelectionRequest``
-launch's ``content_item_return_url`` parameter gives us the URL to POST this
-form submission to. The LMS saves the selection and passes it back to us in the
-launch params whenever this assignment is launched in future.
+ * LMS calls us on `deep_linking_launch`
 
-For more details see the LTI Deep Linking spec:
+    The spec requires that deep linking requests have an ``lti_message_type``
+    identifying the launch as a deep linking one but we don't actually rely
+    on this parameter: instead we use a separate URL
+    to distinguish deep linking launches.
 
-https://www.imsglobal.org/specs/lticiv1p0
+ * In the case of LTI1.3:
 
-Especially this page:
+    * We add configuration to enable a call back to `file_picker_to_formfields_v13`
+    * This provides the form data the front end requires to submit to the LMS
+ * In the case of LTI1.1:
+
+    * The frontend constructs itself the form it needs to send to the LMS
+
+
+For more details see the LTI Deep Linking specs:
+
+- LTI 1.1 https://www.imsglobal.org/specs/lticiv1p0
+
+  Especially this page:
+
+  https://www.imsglobal.org/specs/lticiv1p0/specification-3
+
+- LTI 1.3 https://www.imsglobal.org/spec/lti-dl/v2p0
 
-https://www.imsglobal.org/specs/lticiv1p0/specification-3
 
 Canvas LMS's Content Item docs are also useful:
 
 https://canvas.instructure.com/doc/api/file.content_item.html
```

Comment:
I think this is more relevant for the reader than before. Formatting here is a bit inconsistent

Reply:
Moved things around a little bit, I think it looks a bit better now.